File: apps/cn/game-backend/apps/game/v2/models/insights.go
package models

import "time"

type InsightMetric struct {
	Key           string   `json:"key"`
	AsOf          string   `json:"as_of"`
	Value         *float64 `json:"value"`
	Delta30D      *float64 `json:"delta_30d"`
	Coverage      *float64 `json:"coverage"`
	Known         int64    `json:"known"`
	Eligible      int64    `json:"eligible"`
	AvailableFrom *string  `json:"available_from"`
}

type InsightTrendPoint struct {
	Date     string   `json:"date"`
	Value    *float64 `json:"value"`
	Coverage *float64 `json:"coverage"`
}

type InsightMetricTrend struct {
	Key              string              `json:"key"`
	RequestedRange   string              `json:"requested_range"`
	AvailableFrom    *string             `json:"available_from"`
	AvailableThrough *string             `json:"available_through"`
	Points           []InsightTrendPoint `json:"points"`
}

type InsightDimensionSlice struct {
	Value       string   `json:"value"`
	Label       *string  `json:"label"`
	LabelEn     *string  `json:"label_en"`
	Population  int64    `json:"population"`
	Eligible    int64    `json:"eligible"`
	Known       int64    `json:"known"`
	MetricValue *float64 `json:"metric_value"`
	Coverage    *float64 `json:"coverage"`
}

type InsightDimensionBreakdown struct {
	Key       string                  `json:"key"`
	Dimension string                  `json:"dimension"`
	SliceMode string                  `json:"slice_mode"`
	AsOf      *string                 `json:"as_of"`
	Items     []InsightDimensionSlice `json:"items"`
}

type InsightDimensionSliceRef struct {
	Value   string  `json:"value"`
	Label   *string `json:"label"`
	LabelEn *string `json:"label_en"`
}

type InsightDimensionTrendPoint struct {
	Date        string   `json:"date"`
	Population  int64    `json:"population"`
	Eligible    int64    `json:"eligible"`
	Known       int64    `json:"known"`
	MetricValue *float64 `json:"metric_value"`
	Coverage    *float64 `json:"coverage"`
}

type InsightDimensionTrend struct {
	Key              string                       `json:"key"`
	Dimension        string                       `json:"dimension"`
	Slice            InsightDimensionSliceRef     `json:"slice"`
	SliceMode        string                       `json:"slice_mode"`
	RequestedRange   string                       `json:"requested_range"`
	AsOf             *string                      `json:"as_of"`
	AvailableFrom    *string                      `json:"available_from"`
	AvailableThrough *string                      `json:"available_through"`
	Points           []InsightDimensionTrendPoint `json:"points"`
}

type InsightEntityRef struct {
	ID   int64  `json:"id"`
	Name string `json:"name"`
}

type InsightChange struct {
	Type       string           `json:"type"`
	Date       string           `json:"date"`
	OccurredAt *time.Time       `json:"occurred_at"`
	Entity     InsightEntityRef `json:"entity"`
	Detail     any              `json:"detail"`
}

type InsightExplorerChange struct {
	Domain     string           `json:"domain"`
	Category   string           `json:"category"`
	Type       string           `json:"type"`
	Date       string           `json:"date"`
	OccurredAt *time.Time       `json:"occurred_at"`
	Entity     InsightEntityRef `json:"entity"`
	Detail     any              `json:"detail"`
}

type InsightChangeExplorerPage struct {
	Items      []InsightExplorerChange `json:"items"`
	NextCursor *string                 `json:"next_cursor"`
}

type InsightChangeExplorerQuery struct {
	Range    string
	Category string
	Type     string
	Cursor   string
	Limit    int32
}

type InsightOverview struct {
	GeneratedAt   time.Time       `json:"generated_at"`
	EntityCount   int64           `json:"entity_count"`
	Changes7D     int64           `json:"changes_7d"`
	Metrics       []InsightMetric `json:"metrics"`
	RecentChanges []InsightChange `json:"recent_changes"`
}

type InsightGameState struct {
	Free    *bool   `json:"free"`
	Windows *bool   `json:"windows"`
	Mac     *bool   `json:"mac"`
	Linux   *bool   `json:"linux"`
	Release *string `json:"release"`
	AsOf    *string `json:"as_of"`
}

type InsightPlayerSummary struct {
	Current              *int64     `json:"current"`
	Peak30D              *int64     `json:"peak_30d"`
	Average30D           *float64   `json:"average_30d"`
	AsOf                 *time.Time `json:"as_of"`
	FactThrough          *string    `json:"fact_through"`
	EligibleFrom30D      *string    `json:"eligible_from_30d"`
	ObservedDays30D      int64      `json:"observed_days_30d"`
	SuccessfulSamples30D int64      `json:"successful_samples_30d"`
	SampleCoverage30D    *float64   `json:"sample_coverage_30d"`
}

type InsightPrice struct {
	Region          string  `json:"region"`
	State           string  `json:"state"`
	Currency        *string `json:"currency"`
	InitialAmount   *int64  `json:"initial_amount"`
	FinalAmount     *int64  `json:"final_amount"`
	DiscountPercent *int32  `json:"discount_percent"`
	AsOf            string  `json:"as_of"`
}

type InsightObservedLow struct {
	Amount          int64  `json:"amount"`
	Currency        string `json:"currency"`
	FirstSeen       string `json:"first_seen"`
	ObservedSince   string `json:"observed_since"`
	InitialAmount   int64  `json:"initial_amount"`
	DiscountPercent int32  `json:"discount_percent"`
}

type InsightRegionalPrice struct {
	Region          string              `json:"region"`
	Available       bool                `json:"available"`
	State           *string             `json:"state"`
	Currency        *string             `json:"currency"`
	InitialAmount   *int64              `json:"initial_amount"`
	FinalAmount     *int64              `json:"final_amount"`
	DiscountPercent *int32              `json:"discount_percent"`
	ObservedLow     *InsightObservedLow `json:"observed_low"`
}

type InsightRegionalPrices struct {
	AsOf    *string                `json:"as_of"`
	Regions []InsightRegionalPrice `json:"regions"`
}

type GameInsights struct {
	Game           InsightEntityRef      `json:"game"`
	State          InsightGameState      `json:"state"`
	Players        InsightPlayerSummary  `json:"players"`
	Price          *InsightPrice         `json:"price"`
	RegionalPrices InsightRegionalPrices `json:"regional_prices"`
	RecentChanges  []InsightChange       `json:"recent_changes"`
}

type InsightPlayerPoint struct {
	Date string   `json:"date"`
	Min  *int64   `json:"min"`
	Max  int64    `json:"max"`
	Avg  *float64 `json:"avg"`
}

type InsightPlayerHistory struct {
	RequestedRange   string               `json:"requested_range"`
	AvailableFrom    *string              `json:"available_from"`
	AvailableThrough *string              `json:"available_through"`
	Points           []InsightPlayerPoint `json:"points"`
}

type InsightPricePoint struct {
	Date            string  `json:"date"`
	State           string  `json:"state"`
	Currency        *string `json:"currency"`
	InitialAmount   *int64  `json:"initial_amount"`
	FinalAmount     *int64  `json:"final_amount"`
	DiscountPercent *int32  `json:"discount_percent"`
}

type InsightPriceHistory struct {
	Region           string              `json:"region"`
	RequestedRange   string              `json:"requested_range"`
	AvailableFrom    *string             `json:"available_from"`
	AvailableThrough *string             `json:"available_through"`
	Points           []InsightPricePoint `json:"points"`
}

type InsightPlayerRankingItem struct {
	Rank              int32            `json:"rank"`
	Game              InsightEntityRef `json:"game"`
	Value             float64          `json:"value"`
	ObservedAt        *time.Time       `json:"observed_at"`
	EligibleFrom      *string          `json:"eligible_from"`
	ObservedDays      *int64           `json:"observed_days"`
	SuccessfulSamples *int64           `json:"successful_samples"`
	SampleCoverage    *float64         `json:"sample_coverage"`
}

type InsightPlayerRanking struct {
	Metric                 string                     `json:"metric"`
	Basis                  string                     `json:"basis"`
	SnapshotScheduledFor   *time.Time                 `json:"snapshot_scheduled_for"`
	LatestSlotScheduledFor *time.Time                 `json:"latest_slot_scheduled_for"`
	ObservedFrom           *time.Time                 `json:"observed_from"`
	ObservedThrough        *time.Time                 `json:"observed_through"`
	WindowFrom             *string                    `json:"window_from"`
	WindowThrough          *string                    `json:"window_through"`
	Population             int64                      `json:"population"`
	Ranked                 int64                      `json:"ranked"`
	EntityCoverage         *float64                   `json:"entity_coverage"`
	Items                  []InsightPlayerRankingItem `json:"items"`
}

type InsightPlayerRankingQuery struct {
	Metric string
	Limit  int32
}

type InsightPriceOverview struct {
	Region          string   `json:"region"`
	AsOf            *string  `json:"as_of"`
	Population      int64    `json:"population"`
	Priced          int64    `json:"priced"`
	Free            int64    `json:"free"`
	Unpriced        int64    `json:"unpriced"`
	Unknown         int64    `json:"unknown"`
	Unavailable     int64    `json:"unavailable"`
	Known           int64    `json:"known"`
	Coverage        *float64 `json:"coverage"`
	Discounted      int64    `json:"discounted"`
	DiscountedShare *float64 `json:"discounted_share"`
}

type InsightDiscountItem struct {
	Game            InsightEntityRef    `json:"game"`
	Currency        string              `json:"currency"`
	InitialAmount   int64               `json:"initial_amount"`
	FinalAmount     int64               `json:"final_amount"`
	DiscountPercent int32               `json:"discount_percent"`
	ObservedLow     *InsightObservedLow `json:"observed_low"`
}
type InsightDiscounts struct {
	Region string                `json:"region"`
	AsOf   *string               `json:"as_of"`
	Items  []InsightDiscountItem `json:"items"`
}

type InsightLanguageItem struct {
	Code                   string   `json:"code"`
	SteamName              string   `json:"steam_name"`
	SupportedGames         int64    `json:"supported_games"`
	Share                  *float64 `json:"share"`
	ExplicitFullAudioGames int64    `json:"explicit_full_audio_games"`
	ExplicitFullAudioShare *float64 `json:"explicit_full_audio_share"`
}
type InsightLanguageOverview struct {
	AsOf                  *string               `json:"as_of"`
	FreshnessSeconds      int64                 `json:"freshness_seconds"`
	Population            int64                 `json:"population"`
	Fresh                 int64                 `json:"fresh"`
	Stale                 int64                 `json:"stale"`
	Unobserved            int64                 `json:"unobserved"`
	Coverage              *float64              `json:"coverage"`
	FullyNormalizedGames  int64                 `json:"fully_normalized_games"`
	UnmappedGames         int64                 `json:"unmapped_games"`
	UnmappedEntries       int64                 `json:"unmapped_entries"`
	NormalizationCoverage *float64              `json:"normalization_coverage"`
	Items                 []InsightLanguageItem `json:"items"`
}

type InsightMetricContract struct {
	PublicKey   string
	InternalKey string
	Version     int32
}

type InsightDimensionContract struct {
	PublicKey   string
	InternalKey string
	SliceMode   string
}

type InsightDimensionRecord struct {
	Value         string
	Label         *string
	LabelEn       *string
	Population    int64
	Eligible      int64
	PositiveCount int64
	NegativeCount int64
}

type InsightDimensionAvailabilityRecord struct {
	Label            *string
	LabelEn          *string
	AvailableFrom    *time.Time
	AvailableThrough *time.Time
}

type InsightDimensionTrendRecord struct {
	FactDate      time.Time
	Population    int64
	Eligible      int64
	PositiveCount int64
	NegativeCount int64
}

type InsightMetricSummaryRecord struct {
	FactDate              time.Time
	EligibleCount         int64
	PositiveCount         int64
	NegativeCount         int64
	PreviousPositiveCount *int64
	PreviousNegativeCount *int64
	AvailableFrom         *time.Time
}

type InsightMetricTrendRecord struct {
	FactDate      time.Time
	EligibleCount int64
	PositiveCount int64
	NegativeCount int64
}

type InsightGameRecord struct {
	ID   int64
	Name string
}

type InsightGameStateRecord struct {
	GameID           int64
	FactDate         time.Time
	TrackingPeriodID int64
	AppID            int64
	Free             *bool
	Windows          *bool
	Mac              *bool
	Linux            *bool
	Release          *string
}

type InsightPlayerSummaryRecord struct {
	HasCurrent        bool
	Current           int64
	CurrentAt         *time.Time
	Peak30D           *int64
	Average30D        *float64
	FactThrough       *time.Time
	EligibleFrom      *time.Time
	ObservedDays      int64
	SuccessfulSamples int64
	SampleCoverage    *float64
}

type InsightPriceRecord struct {
	FactDate        time.Time
	State           string
	Currency        *string
	InitialAmount   *int64
	FinalAmount     *int64
	DiscountPercent *int32
}

type InsightRegionalPriceRecord struct {
	Region          string
	Available       bool
	FactDate        time.Time
	State           *string
	Currency        *string
	InitialAmount   *int64
	FinalAmount     *int64
	DiscountPercent *int32
}
type InsightObservedLowRecord struct {
	Amount          int64
	Currency        string
	FirstSeen       time.Time
	ObservedSince   time.Time
	InitialAmount   int64
	DiscountPercent int32
}
type InsightPlayerRankingMetaRecord struct {
	SnapshotScheduledFor   *time.Time
	LatestSlotScheduledFor *time.Time
	ObservedFrom           *time.Time
	ObservedThrough        *time.Time
	WindowFrom             *time.Time
	WindowThrough          *time.Time
	Population             int64
	Ranked                 int64
}
type InsightPlayerRankingRecord struct {
	GameID            int64
	GameName          string
	Value             float64
	ObservedAt        *time.Time
	EligibleFrom      *time.Time
	ObservedDays      *int64
	SuccessfulSamples *int64
	SampleCoverage    *float64
}
type InsightPriceOverviewRecord struct {
	AsOf                                                                 *time.Time
	Population, Priced, Free, Unpriced, Unknown, Unavailable, Discounted int64
}
type InsightDiscountRecord struct {
	AsOf                       time.Time
	GameID, TrackingPeriodID   int64
	GameName, Currency         string
	InitialAmount, FinalAmount int64
	DiscountPercent            int32
}
type InsightLanguageOverviewRecord struct {
	AsOf                                                                                       *time.Time
	Population, Fresh, Stale, Unobserved, FullyNormalizedGames, UnmappedGames, UnmappedEntries int64
}
type InsightLanguageRecord struct {
	Code, SteamName                        string
	SupportedGames, ExplicitFullAudioGames int64
}

type InsightPlayerPointRecord struct {
	FactDate time.Time
	Min      *int64
	Max      int64
	Avg      *float64
}

type InsightChangeRecord struct {
	EntityID        int64
	EntityName      string
	DetectorKey     string
	DetectorVersion int32
	EventCode       string
	ProjectionDate  time.Time
	TimeBasis       string
	EventAt         *time.Time
	PrecisionRank   int32
	EventSortAt     time.Time
	OpaqueTie       string
}

type InsightChangeExplorerPosition struct {
	ProjectionDate time.Time
	PrecisionRank  int32
	EventSortAt    time.Time
	OpaqueTie      string
}

type InsightChangeExplorerConditions struct {
	DetectorKeys []string
	ContractIDs  []string
	RangeThrough time.Time
	RangeDays    int32
	Position     *InsightChangeExplorerPosition
	Limit        int32
}

File: apps/cn/nav-web/app/types/insights.ts
export type InsightRange = '30d' | '90d' | 'all'
export type InsightChangeRange = '7d' | InsightRange
export type NavInsightMetricKey = 'ipv6' | 'tls13' | 'http2' | 'hsts' | 'csp' | 'security_txt' | 'certificate_verified'
export type GameInsightMetricKey = 'free' | 'windows' | 'mac' | 'linux'
export type InsightMetricKey = NavInsightMetricKey | GameInsightMetricKey
export type InsightDomain = 'site' | 'game'
export type SiteInsightDimension = 'country' | 'group' | 'nsfw' | 'public_interest'
export type GameInsightDimension = 'primary_tag' | 'tag'
export type InsightDimension = SiteInsightDimension | GameInsightDimension
export type InsightSliceMode = 'partition' | 'overlapping'
export type SiteInsightChangeCategory = 'capability' | 'target' | 'certificate'
export type GameInsightChangeCategory = 'pricing_model' | 'platform' | 'release' | 'price' | 'discount'
export type InsightChangeCategory = SiteInsightChangeCategory | GameInsightChangeCategory
export type SiteInsightCapabilityKey = NavInsightMetricKey
export type SiteInsightCapabilityState =
  | 'supported'
  | 'unsupported'
  | 'stale'
  | 'not_probed'
  | 'unavailable'
  | 'unknown'
  | 'not_applicable'
export type GameInsightPriceState = 'free' | 'priced' | 'unknown' | 'unpriced'
export type GameInsightRegion = 'CN' | 'US' | 'HK'
export type GamePlayerRankingMetric = 'latest_observed' | 'peak_30d' | 'average_30d'

export interface InsightMetric {
  key: InsightMetricKey
  as_of: string
  value: number | null
  delta_30d: number | null
  coverage: number | null
  known: number
  eligible: number
  available_from: string | null
}

export interface InsightTrendPoint {
  date: string
  value: number | null
  coverage: number | null
}

export interface InsightMetricTrend {
  key: InsightMetricKey
  requested_range: InsightRange
  available_from: string | null
  available_through: string | null
  points: InsightTrendPoint[]
}

export interface InsightDimensionSlice {
  value: string
  label: string | null
  label_en: string | null
  population: number
  eligible: number
  known: number
  metric_value: number | null
  coverage: number | null
}

export interface InsightDimensionBreakdown {
  key: InsightMetricKey
  dimension: InsightDimension
  slice_mode: InsightSliceMode
  as_of: string | null
  items: InsightDimensionSlice[]
}

export interface InsightDimensionSliceRef {
  value: string
  label: string | null
  label_en: string | null
}

export interface InsightDimensionTrendPoint {
  date: string
  population: number
  eligible: number
  known: number
  metric_value: number | null
  coverage: number | null
}

export interface InsightDimensionTrend {
  key: InsightMetricKey
  dimension: InsightDimension
  slice: InsightDimensionSliceRef
  slice_mode: InsightSliceMode
  requested_range: InsightRange
  as_of: string | null
  available_from: string | null
  available_through: string | null
  points: InsightDimensionTrendPoint[]
}

export interface InsightEntityRef {
  id: number
  name: string
}

export interface InsightChange {
  type: string
  date: string
  occurred_at: string | null
  entity: InsightEntityRef
  detail: unknown | null
}

export interface InsightOverview {
  generated_at: string
  entity_count: number
  changes_7d: number
  metrics: InsightMetric[]
  recent_changes: InsightChange[]
}

export interface InsightFeedItem extends InsightChange {
  domain: InsightDomain
}

export interface InsightExplorerChange extends InsightChange {
  domain: InsightDomain
  category: InsightChangeCategory
}

export interface InsightChangeExplorerPage {
  items: InsightExplorerChange[]
  next_cursor: string | null
}

export interface SiteInsightCapability {
  key: SiteInsightCapabilityKey
  as_of: string | null
  state: SiteInsightCapabilityState
  ecosystem: {
    value: number | null
    coverage: number | null
  }
}

export interface SiteInsights {
  site: InsightEntityRef
  capabilities: SiteInsightCapability[]
  recent_changes: InsightChange[]
}

export interface SiteCompareCapability {
  key: SiteInsightCapabilityKey
  state: SiteInsightCapabilityState
}

export interface SiteCompareCertificate {
  target: string
  not_after: string | null
  days_to_expiry: number | null
  expiry_status: CertificateExpiryStatus | null
  verified: boolean | null
  verification_issue: CertificateVerificationIssue | null
  issuer: string | null
  observed_at: string | null
}

export interface SiteCompareItem {
  site: InsightEntityRef
  capabilities: SiteCompareCapability[]
  certificate: SiteCompareCertificate | null
}

export interface SiteCompare {
  status: 'ready' | 'insufficient_data'
  as_of: string | null
  sites: SiteCompareItem[]
}

export type CertificateExpiryStatus = 'expired' | 'expires_within_7d' | 'expires_in_8_30d' | 'later'
export type CertificateVerificationIssue = 'hostname_mismatch' | 'unknown_authority' | 'expired' | 'not_yet_valid' | 'incompatible_usage' | 'other'

export interface CertificateInsightItem {
  site: InsightEntityRef
  target: string
  not_after: string | null
  days_to_expiry: number | null
  expiry_status: CertificateExpiryStatus | null
  verified: boolean | null
  verification_issue: CertificateVerificationIssue | null
  issuer: string | null
  observed_at: string | null
}

export interface CertificateInsightOverview {
  as_of: string | null
  reference_at: string | null
  freshness_seconds: number
  population: number
  eligible: number
  verification: { known: number; verified: number; failed: number; coverage: number | null }
  quality: { not_applicable: number; stale: number; not_probed: number; probe_failed: number; unknown: number }
  expiry: {
    known: number
    coverage: number | null
    expired: number
    expires_within_7d: number
    expires_in_8_30d: number
    later: number
  }
  expiry_attention: CertificateInsightItem[]
  verification_issues: CertificateInsightItem[]
}

export interface GameInsightState {
  free: boolean | null
  windows: boolean | null
  mac: boolean | null
  linux: boolean | null
  release: string | null
  as_of: string | null
}

export interface GameInsightPlayerSummary {
  current: number | null
  peak_30d: number | null
  average_30d: number | null
  as_of: string | null
  fact_through: string | null
  eligible_from_30d: string | null
  observed_days_30d: number
  successful_samples_30d: number
  sample_coverage_30d: number | null
}

export interface GameInsightPrice {
  region: 'CN'
  state: GameInsightPriceState
  currency: string | null
  initial_amount: number | null
  final_amount: number | null
  discount_percent: number | null
  as_of: string | null
}

export interface GameInsightObservedLow {
  amount: number
  currency: string
  first_seen: string
  observed_since: string
  initial_amount: number
  discount_percent: number
}

export interface GameInsightRegionalPrice {
  region: GameInsightRegion
  available: boolean
  state: GameInsightPriceState | null
  currency: string | null
  initial_amount: number | null
  final_amount: number | null
  discount_percent: number | null
  observed_low: GameInsightObservedLow | null
}

export interface GameInsightRegionalPrices {
  as_of: string | null
  regions: GameInsightRegionalPrice[]
}

export interface GameInsights {
  game: InsightEntityRef
  state: GameInsightState
  players: GameInsightPlayerSummary
  price: GameInsightPrice | null
  regional_prices: GameInsightRegionalPrices
  recent_changes: InsightChange[]
}

export interface GameComparePlayers {
  current_available: boolean
  current: number | null
  observed_at: string | null
  peak_30d: number | null
  average_30d: number | null
  eligible_from_30d: string | null
  observed_days_30d: number
  successful_samples_30d: number
  sample_coverage_30d: number | null
}

export interface GameCompareLanguages {
  evidence: 'fresh' | 'stale' | 'unobserved'
  supported: string[]
  explicit_full_audio: string[]
  unknown_names: string[]
}

export interface GameCompareItem {
  game: InsightEntityRef
  state: Omit<GameInsightState, 'as_of'>
  players: GameComparePlayers
  price: GameInsightRegionalPrice
  languages: GameCompareLanguages
}

export interface GameCompare {
  status: 'ready' | 'insufficient_data'
  region: GameInsightRegion
  state_as_of: string | null
  player_snapshot_scheduled_for: string | null
  player_fact_through: string | null
  games: GameCompareItem[]
}

export interface GameInsightPlayerPoint {
  date: string
  min: number | null
  max: number
  avg: number | null
}

export interface GameInsightPlayerHistory {
  requested_range: InsightRange
  available_from: string | null
  available_through: string | null
  points: GameInsightPlayerPoint[]
}

export interface GameInsightPricePoint {
  date: string
  state: GameInsightPriceState
  currency: string | null
  initial_amount: number | null
  final_amount: number | null
  discount_percent: number | null
}

export interface GameInsightPriceHistory {
  region: GameInsightRegion
  requested_range: InsightRange
  available_from: string | null
  available_through: string | null
  points: GameInsightPricePoint[]
}

export interface GamePlayerRankingItem {
  rank: number
  game: InsightEntityRef
  value: number
  observed_at: string | null
  eligible_from: string | null
  observed_days: number | null
  successful_samples: number | null
  sample_coverage: number | null
}

export interface GamePlayerRanking {
  metric: GamePlayerRankingMetric
  basis: 'scheduled_snapshot' | 'finalized_daily_facts'
  snapshot_scheduled_for: string | null
  latest_slot_scheduled_for: string | null
  observed_from: string | null
  observed_through: string | null
  window_from: string | null
  window_through: string | null
  population: number
  ranked: number
  entity_coverage: number | null
  items: GamePlayerRankingItem[]
}

export interface GamePriceOverview {
  region: GameInsightRegion
  as_of: string | null
  population: number
  priced: number
  free: number
  unpriced: number
  unknown: number
  unavailable: number
  known: number
  coverage: number | null
  discounted: number
  discounted_share: number | null
}

export interface GameDiscountItem {
  game: InsightEntityRef
  currency: string
  initial_amount: number
  final_amount: number
  discount_percent: number
  observed_low: GameInsightObservedLow | null
}
export interface GameDiscounts { region: GameInsightRegion; as_of: string | null; items: GameDiscountItem[] }

export interface GameLanguageItem {
  code: string
  steam_name: string
  supported_games: number
  share: number | null
  explicit_full_audio_games: number
  explicit_full_audio_share: number | null
}
export interface GameLanguageOverview {
  as_of: string | null
  freshness_seconds: number
  population: number
  fresh: number
  stale: number
  unobserved: number
  coverage: number | null
  fully_normalized_games: number
  unmapped_games: number
  unmapped_entries: number
  normalization_coverage: number | null
  items: GameLanguageItem[]
}
